Initial release for Jaeger Helm Chart

davidvonthenen opened the pull request:
This is the initial release for the Jaeger Helm Chart.

helm lint: Pass
helm install .: Works with the minimum required resources
data persistence: Done in the cassandra subchart
README.md: Included
NOTES.txt: Included
Best practices labels and values: Yes, with exception. Some of the values override values in the cassandra chart. Those keys do not follow the lowercase/camelcase convention.

Commit:
Initial release for Jaeger Helm Chart

unguiculus commented:
Thanks for the PR. Please check our best practices for standard labels, especially regarding the `app` label.

https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/labels.md

davidvonthenen commented:
@unguiculus Will check it out. Thanks!

Commit:
Update app label to match best practices

davidvonthenen commented:
Hi @unguiculus I updated the app label to match best practices and verified that the other labels comply as well.

FYI my response might be delayed as I am attending Open Source Summit NA and MesosCon NA this week.

unguiculus commented on incubator/jaeger/Chart.yaml:
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+description: A Jaeger Helm chart for Kubernetes
+name: jaeger
+version: 0.1.0
```
Add `appVersion`

davidvonthenen replied:
Added.

unguiculus commented on incubator/jaeger/templates/_helpers.tpl:
```diff
@@ -0,0 +1,28 @@
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "jaeger" -}}
```
Call it `jaeger.name`

davidvonthenen replied:
Changed to jaeger.name

unguiculus commented on incubator/jaeger/templates/agent-ds.yaml:
```diff
@@ -0,0 +1,38 @@
+    app: "{{ template "jaeger" . }}"
+    jaeger-infra: agent-daemonset
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    component: "{{ template "jaeger.fullname" . }}-agent"
```
Don't add the fullname to the component name. Should just be `agent`.

davidvonthenen replied:
I mirrored the component names based on what is recommended from the Jaeger repo for Kubernetes. Their best practice is to have jaeger-agent as seen here:
https://github.com/jaegertracing/jaeger-kubernetes/blob/master/production/jaeger-production-template.yml

I can make the component name static meaning replace "{{ template "jaeger.fullname" . }}-agent" to "jaeger-agent".

davidvonthenen replied:
Changed to static names.

unguiculus commented on incubator/jaeger/templates/agent-svc.yaml:
```diff
@@ -0,0 +1,32 @@
+    protocol: UDP
+    targetPort: {{ .Values.service.agent.binaryPort }}
+  clusterIP: None
+  selector:
```
Selector should have `app`, `component`, and `release` labels.

davidvonthenen replied:
Added app, component, and release to selector.

unguiculus commented on incubator/jaeger/templates/collector-deploy.yaml:
```diff
@@ -0,0 +1,45 @@
+  template:
+    metadata:
+      labels:
+        jaeger-infra: collector-pod
```
`app`, `component`, and `release` labels should go here.

davidvonthenen replied:
Added app, component, and release.

unguiculus commented on incubator/jaeger/values.yaml:
```diff
@@ -0,0 +1,79 @@
+# Default values for jaeger.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
```
We usually group by component instead of resource type. I think that would make more sense.

davidvonthenen replied:
Looking to refactor for this now.

davidvonthenen replied:
Refactored grouping by component. This is completed.

unguiculus commented on incubator/jaeger/values.yaml:
```diff
@@ -0,0 +1,79 @@
+  persistence:
+    enabled: false
+  config:
+    cluster_name: jaeger
```
Camel case, please.

davidvonthenen replied:
I am overwriting values in the Cassandra chart and that is how they appear. What's the best way to resolve this?

davidvonthenen replied:
@unguiculus requesting feedback for the above question

davidvonthenen replied:
I can always submit a PR to fix their camel case issue.

unguiculus reviewed (changes requested):
I made a few comments. Please apply them throughout the whole chart. I usually only specified them once.

Please add `resources` and `nodeSelector`.

Here's a good example: https://github.com/kubernetes/charts/tree/master/stable/nginx-ingress
You may take this also as reference for grouping values.

Commit:
Added in changes based on the following review
https://github.com/kubernetes/charts/pull/2078/files/eccf2e7f224a3519e128a70242613ba6c0535e8a

Commit:
Place service related values into a subgroup called service in the values.yaml

Commit:
Add back in the app, component, and release labels with the exception of the cassandra schema Job

Commit:
Fix query-ui service type

Commit:
Fix the resources for various components

davidvonthenen commented:
@unguiculus  Added in resources and nodeSelector for the various components.

Also updated the various files based on the review at: https://github.com/kubernetes/charts/pull/2078/files/eccf2e7f224a3519e128a70242613ba6c0535e8a

The last outstanding item about the Cassandra subchart not using camel case. Looking for guidance on how to proceed.

Commit:
Fix jaeger-agent nodes pointing to the correct collector FQDN

davidvonthenen commented:
Made a small correction to the jaeger-agent pointing to the correct FQDN of the jaeger-collector. Tested and working.

Just waiting for feedback on the Cassandra helm chart not using camel case.

Commit:
Add a comment about value grouping and Cassandra subchart overrides.

unguiculus commented:
I overlooked that fact that the camel case is for Cassandra. It's fine then. you have no choice.

unguiculus commented on incubator/jaeger/templates/agent-ds.yaml:
```diff
@@ -0,0 +1,45 @@
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
```
I guess I confused you. Maybe I was not clear enough. Sorry. I did not want you to change the names, only the component labels. Do it like so throughout the chart:

`name: "{{ template "jaeger.fullname" . }}-agent"`

`component: agent`

davidvonthenen replied:
Ahhh I see. Will provide an update shortly.

davidvonthenen replied:
This has been addressed.

Commit:
Update name on components and component names

unguiculus commented on incubator/jaeger/templates/agent-ds.yaml:
```diff
@@ -18,14 +18,14 @@ spec:
 {{ toYaml .Values.agent.nodeSelector | indent 8 }}
       containers:
-      - name: "jaeger-agent"
+      - name: "{{ template "jaeger.fullname" . }}-agent"
```
You could have left the container names as they were. Feel free to change those back if you like.

davidvonthenen replied:
I would like to keep them with the jaeger.fullname so that if there is a situation where you want to deploy multiple independent instances, this gives you that flexibility for unique names.

unguiculus reviewed (changes requested):
/ok-to-test

Commit:
Update the command line to use fullname

unguiculus commented:
/ok-to-test

davidvonthenen commented:
@unguiculus  I believe that fixes all outstanding issues. Please provide feedback if you see any other issues. I have tested this as well and all seems to function as expected.

unguiculus commented:
Only my question regarding the container names is still open (https://github.com/kubernetes/charts/pull/2078#discussion_r141441151). If you like to keep them as they are now, I'll merge the PR.

davidvonthenen commented:
@unguiculus I replied to your comment above. Would like to keep the jaeger.fullname. Thanks for all your help in this! I really appreciate it!

unguiculus commented:
/lgtm
